### ISSUE TITLE
name clash in twisted.protocols.ftp

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10357]
|<img alt="fdrake's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| fdrake reported|
|-|-|
|Trac ID|trac#430|
|Type|defect|
|Created|2003-12-12 23:04:20Z|

Attachments:

* [ftp-2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f8c/f8c024c4ad95bf78baaf9d88334722b84f8a930b/b023f1c7195740242c59dc62bbe0687e9c80707b.diff) (955 bytes) - added by fdrake on 2003-12-12 23:26:12Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__430 430
type__defect defect
reporter__fdrake fdrake
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ftp ftp
keywords__ 
time__1071270260000000 1071270260000000
changetime__1469617430156211 1469617430156211
version__ 
owner__ 
cc__fdrake
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="fdrake's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fdrake commented|
|-|-|

```
#!html
<pre>
The twisted.protocols.ftp module imports two modules named "error" into the
module namespace.  One is from the twisted.cred package, and the other is from
twisted.internet.  Since the error module from cred is imported last, it wins,
and checks using twisted.internet errors cause AttributeError to be raised since
the exceptions being checked don't exist in t.c.error.

Sorry, no patch just yet.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="fdrake's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fdrake commented|
|-|-|

```
#!html
<pre>
This patch seems sufficient, but requires Python 2.0 or newer; I'm not sure what
versions of Python Twisted is intended to support.  (The Python version issue is
that this uses "from twisted.cred import error as cred_error" to rename the
imported module as part of the import.)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="fdrake's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fdrake commented|
|-|-|

```
#!html
<pre>
It turns out that this was already reported as Issue428, but the patch provided
in this issue has been tested.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
fdrake: Twisted requires Python 2.2 or greater; this is documented in the README.

I've applied your patch to CVS:
Checking in ftp.py;
/cvs/Twisted/twisted/protocols/ftp.py,v  &lt;--  ftp.py
new revision: 1.99; previous revision: 1.98
done

Thanks!
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub commented|
|-|-|

In changeset 26b6d334f3eb57bff5f64a3180855c200d9c2223
```
#!CommitTicketReference repository="" revision="26b6d334f3eb57bff5f64a3180855c200d9c2223"
Merge pull request #430 from twisted/8693-anchat1990-twisted-tls-hugeWrite-error-1

Author: anchat1990
Reviewer: rodrigc
Fixes: #8693

Updated the tests to reflect that hugeWrite fails when using TLSv1_1
```
